weatherman.py

```diff
@@ -99,71 +87,91 @@ def execute_e_argument(my_weatherlist):
     else:
         humidity_date_str = "N/A"
 
-    print(f'Highest: {global_max_temp} on {max_temp_str}')
-    print(f'Lowest: {global_min_temp} on {min_temp_str}')
-    print(f'Humidity: {humidity_percentage} on {humidity_date_str}')
+    print(f'Highest: {round(global_max_temp)}C on {max_temp_str}')
+    print(f'Lowest: {round(global_min_temp)}C on {min_temp_str}')
+    print(f'Humidity: {round(highest_humidity)} on {humidity_date_str}')
 
 def execute_a_argument(my_weatherlist):
     temp_dict = {}
+
     for line in my_weatherlist:
         try:
             date_str = line[0]
-            temperature = float(line[1])
-            humidity = float(line[-1])
+            max_temperature = float(line[1])
+            min_temperature = float(line[3])
+            humidity = float(line[8])
             date_obj = datetime.datetime.strptime(date_str, "%Y-%m-%d")
             year = date_obj.year
 
             if year not in temp_dict:
-                temp_dict[year] = {'temp' : [], 'humidities' : []}
+                temp_dict[year] = {'temp1' : [],'temp2' : [], 'humidities' : []}
```

Comment:
As you are handling only one year in one command, so I think that the year key unnecessary here.